### ISSUE TITLE
PhantomJS crashing/asynchronous script timeout with ChromeDriver

### ISSUE THREAD

**Issue by workflow** [xolvio/meteor-cucumber#107]
First of all, thank you for this amazing package!

I get the following PhantomJS crash

```
[chimp][phantom.stderr] PhantomJS has crashed. Please read the crash reporting guide at https://github.com/ariya/phantomjs/wiki/Crash-Reporting and file a bug report at https://github.com/ariya/phantomjs/issues/new with the crash dump file attached: /tmp/1e377fe0-4516-1339-7f367d91-4d466e2b.dmp
```

...when trying to run the following Step more than once (creating a user server-side and then logging in):

```javascript
this.Given(/^I am an admin$/, function(callback) {
    this.mirror.call('createAdminUser').then(function() {
        browser.url(url.resolve(process.env.ROOT_URL, '/')).executeAsync(function(done) {
            Meteor.loginWithPassword('admin', '123', done);
        }).should.eventually.not.have.deep.property('value.message').and.notify(callback);
    });
});

```

Now these are my first steps with cucumber in general so the above code might be complete jibberdang. Is there something fundamentally wrong with it?

Otherwise I'll post the full debug stack for further debugging. Any help or advice greatly appreciated.

**Comment by samhatoum** [comment]
You're welcome!

I suggest trying with `SELENIUM_BROWSER=chrome meteor` and seeing if the problem happens at all. That will tell you if it's an issue with your test code (doesn't look like it) or the app being too complicated for Phantom's little brain to handle :)

I've had to shim phantom in the past with polyfills, but if this is intermittent it probably isn't a shim issue. 

Another thing to try is to logout before logging in maybe?

**Comment by workflow** [comment]
With SELENIUM_BROWSER=chrome I get the following error instead:

```
RuntimeError: RuntimeError
           (ScriptTimeout:28) A script did not complete before its timeout expired.
           Problem: asynchronous script timeout: result was not received in 0 seconds

           Callstack:
           -> executeAsync([Function])
```

In more detail:

```
[chimp][selenium.stderr] 11:58:40.637 INFO - Executing: [execute async script: return (function (done) {
                    Meteor.loginWithPassword('admin', '123', done);
                }).apply(null, arguments);, []])

[chimp][selenium.stderr] 11:58:41.025 WARN - Exception thrown
org.openqa.selenium.TimeoutException: asynchronous script timeout: result was not received in 0 seconds
  (Session info: chrome=43.0.2357.65)
  (Driver info: chromedriver=2.15.322448 (52179c1b310fec1797c81ea9a20326839860b7d3),platform=Linux 3.13.0-53-generic x86_64) (WARNING: The server did not provide any stacktrace information)
Command duration or timeout: 44 milliseconds
Build info: version: '2.45.0', revision: '5017cb8', time: '2015-02-26 23:59:50'
System info: host: 'dropzbox.lo', ip: '127.0.1.1', os.name: 'Linux', os.arch: 'amd64', os.version: '3.13.0-53-generic', java.version: '1.8.0_45'
Driver info: org.openqa.selenium.chrome.ChromeDriver
Capabilities [{applicationCacheEnabled=false, rotatable=false, mobileEmulationEnabled=false, chrome={userDataDir=/tmp/.com.google.Chrome.Gl94cO}, takesHeapSnapshot=true, databaseEnabled=false, handlesAlerts=true, version=43.0.2357.65, platform=LINUX, browserConnectionEnabled=false, nativeEvents=true, acceptSslCerts=true, locationContextEnabled=true, webStorageEnabled=true, browserName=chrome, takesScreenshot=true, javascriptEnabled=true, cssSelectorsEnabled=true}]
Session ID: f33a9f403b9d989e145fe5bc86091528
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:422)
	at org.openqa.selenium.remote.ErrorHandler.createThrowable(ErrorHandler.java:204)
	at org.openqa.selenium.remote.ErrorHandler.throwIfResponseFailed(ErrorHandler.java:156)
	at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:599)
	at org.openqa.selenium.remote.RemoteWebDriver.executeAsyncScript(RemoteWebDriver.java:526)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at org.openqa.selenium.support.events.EventFiringWebDriver$2.invoke(EventFiringWebDriver.java:101)
	at com.sun.proxy.$Proxy1.executeAsyncScript(Unknown Source)
	at org.openqa.selenium.support.events.EventFiringWebDriver.executeAsyncScript(EventFiringWebDriver.java:225)
	at org.openqa.selenium.remote.server.handler.ExecuteAsyncScript.call(ExecuteAsyncScript.java:57)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at org.openqa.selenium.remote.server.DefaultSession$1.run(DefaultSession.java:168)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)

```

This happens on the first scenario execution, before any user is logged in - which makes me think it can't be the logout.

**Comment by samhatoum** [comment]
Try to do what you're doing manually by going to the mirror directly, using a browser console to call the method createAdminUser, then try to login with password from the browser. If that doesn't work, please create a reproducible repo for me and I'll look

**Comment by workflow** [comment]
That's a greatly simple idea for debugging which hadn't crossed my mind. Unfortunately, everything runs fine manually on the mirror.

[Here's the repo] (https://github.com/workflow/meteor-cucumber-issue-107) - thx for helping me out on this!

**Comment by samhatoum** [comment]
this repo works for me, I get the following:

```
[xolvio:cucumber] Cucumber is running
@dev
Feature: Crashing ChromeDriver

  In order to help xolv.io debug this
  As a rookie cucumberisto
  I want to crash the heck out of ChromeDriver


  Scenario: Crash and Burn                # features/crash.feature:8
    Given I am an admin                   # features/crash.feature:10
    When I do something I never get to do # features/crash.feature:12
    Then I should never end up here       # features/crash.feature:14


1 scenario (1 undefined)
3 steps (2 undefined, 1 passed)

You can implement step definitions for undefined steps with these snippets:

this.When(/^I do something I never get to do$/, function (callback) {
  // Write code here that turns the phrase above into concrete actions
  callback.pending();
});
```

What do you get?

**Comment by workflow** [comment]
```
[xolvio:cucumber] Cucumber is running
@dev
Feature: Crashing ChromeDriver

  In order to help xolv.io debug this
  As a rookie cucumberisto
  I want to crash the heck out of ChromeDriver


  Scenario: Crash and Burn                # features/crash.feature:8
    Given I am an admin                   # features/crash.feature:10
      RuntimeError: RuntimeError
           (ScriptTimeout:28) A script did not complete before its timeout expired.
           Problem: asynchronous script timeout: result was not received in 0 seconds

           Callstack:
           -> executeAsync([Function])

    When I do something I never get to do # features/crash.feature:12
    Then I should never end up here       # features/crash.feature:14


(::) failed steps (::)

RuntimeError: RuntimeError
     (ScriptTimeout:28) A script did not complete before its timeout expired.
     Problem: asynchronous script timeout: result was not received in 0 seconds

     Callstack:
     -> executeAsync([Function])


Failing scenarios:
home/../phantomcrash/tests/cucumber/features/crash.feature:8 # Scenario: Crash and Burn

1 scenario (1 failed)
3 steps (1 failed, 2 undefined)

You can implement step definitions for undefined steps with these snippets:

this.When(/^I do something I never get to do$/, function (callback) {
  // Write code here that turns the phrase above into concrete actions
  callback.pending();
});

this.Then(/^I should never end up here$/, function (callback) {
  // Write code here that turns the phrase above into concrete actions
  callback.pending();
});

[chimp] Finished running async processes with errors
```

(when running with `SELENIUM_BROWSER=chrome`)

**Comment by samhatoum** [comment]
ok, I can see it now. I've also removed all calls except the loginWithPassword and that seems to be the culprit. I'll let you know when I've figured it out

**Comment by samhatoum** [comment]
ok, this is now fixed in cucumber 0.8.3

it was trying to callback within 0 seconds. I've added a default of a 10s wait

**Comment by workflow** [comment]
Amazing, Sir! Thank you.
